### ISSUE TITLE
Don't reduce map image size during cropping

### ISSUE THREAD

**Issue by kylecorry31** [kylecorry31/Trail-Sense#1033]
Instead, record the warping and apply them before displaying the map / using the projection

The goal would be that the original image is not modified by Trail Sense unless the user chooses to reduce map resolution.

- Save the point to point transform in the DB
- Apply the transform when looking up the tile (same way as rotation), then crop

AI generated proof of concept: https://github.com/kylecorry31/Trail-Sense/tree/map-transform-poc

Needs a lot of cleanup, domain modeling, validation, etc (i.e. full rewrite), my intent was just to make sure it was possible to do with my current architecture and it is. Outside of the POC both the calibration photo map view and the photo map code need to be cleaned up before implementing this. Wait until after the domain rework as well.

## Plan
- Store the percent coordinates of the cropped corners
- Use the perspective transform (via a Matrix) as a projection wrapper within PhotoMapProjection and before the calibration and rotation is done
- Calibration points are percent coordinates of the original image (same as today)
- Crop the tiles to the cropped image
- More work is needed on the subsample scale imageview. Likely need a custom loader and handling there.

**Comment by jptrzy** [comment]
Could you clarify what did you mean by writing the word **size**:
size in disk space or size in scale (how big image is).

**Comment by kylecorry31** [comment]
@jptrzy I've updated the description of this issue to clarify. By size, I meant both disk space and resolution (though cropping will inherently reduce that).

**Comment by StarTroop** [comment]
I'm wondering if this is relevant to the issue I'm having. I want to import a geospatial pdf, a fairly large (dimensionally) map which is sized 55mb. Whenever I import it, it always gets reduced to about 300kb, becoming unusably blurry. I've also tried making both larger and smaller versions of the same map (from the larger geotiff source, converted with gdal), which all get reduced in size to 300kb once imported.
I'm able to properly open and view these pdfs in my pdf viewer app. The practice geospatial pdfs I made with caltopo seemed to import fine, but they are much smaller to begin with.
Should I open this as a new ticket?

**Comment by kylecorry31** [comment]
> I'm wondering if this is relevant to the issue I'm having. I want to import a geospatial pdf, a fairly large (dimensionally) map which is sized 55mb. Whenever I import it, it always gets reduced to about 300kb, becoming unusably blurry. I've also tried making both larger and smaller versions of the same map (from the larger geotiff source, converted with gdal), which all get reduced in size to 300kb once imported. I'm able to properly open and view these pdfs in my pdf viewer app. The practice geospatial pdfs I made with caltopo seemed to import fine, but they are much smaller to begin with. Should I open this as a new ticket?

Hi, have you tried turning off the reduce photo/pdf resolution settings in photo maps settings? If so, then yes I recommend logging a new issue using the bug template. 

That's assuming you aren't cropping the image during import (which is this issue)

**Comment by StarTroop** [comment]
> 
> Hi, have you tried turning off the reduce photo/pdf resolution settings in photo maps settings? If so, then yes I recommend logging a new issue using the bug template.
> 
> That's assuming you aren't cropping the image during import (which is this issue)

Sorry, I didn't even realise there was such an option. I assumed this was a bug since the automatic reduction of 55mb to 300kb doesn't seem like a sensible default, IMO.
I've disabled the resizing but now the app crashes whenever I try to open my maps, though that's clearly a separate issue.

**Comment by kylecorry31** [comment]
> > Hi, have you tried turning off the reduce photo/pdf resolution settings in photo maps settings? If so, then yes I recommend logging a new issue using the bug template.
> > That's assuming you aren't cropping the image during import (which is this issue)
> 
> Sorry, I didn't even realise there was such an option. I assumed this was a bug since the automatic reduction of 55mb to 300kb doesn't seem like a sensible default, IMO. I've disabled the resizing but now the app crashes whenever I try to open my maps, though that's clearly a separate issue.

Yeah, that's a separate issue likely related to https://github.com/kylecorry31/Trail-Sense/issues/1345 

If you can provide some details in a new bug report that will help me investigate, though it's likely that the image is too large for the device.

I went with a high reduction by default to support lower end devices and for the primary use case of using the camera to import the map. That reduction amount was derived from my testing across various devices.
